src/extractAndWritePOTFromMessagesSync.js

```diff
@@ -12,10 +13,17 @@ const customKeyMapper = (message, messageKey, filename) => ({
 const customKeyMapperFactory = (messageKey = 'defaultMessage') =>
   (message, filename) => customKeyMapper(message, messageKey, filename);
 
-function extractAndWritePOTFromMessagesSync(srcPatterns, { messageKey, output }) {
+function extractAndWritePOTFromMessagesSync(srcPatterns, { messageKey, output, headerOptions }) {
```

Comment:
I've been studying this but it extend the discussion to the point of generating jsDoc for this module. I've seen myself pulling lot of dependencies to generate documentation and I think as i'm not the maintainer that it is not my purporse. I can gladly help once a basic jsdoc support is added to the main branch.

I've also think that headerOptions should be merged with the second argument as a flat options object.

I suggest we wait for the merge to occur, and that you define jsdoc rules and implementation. Then i'll amend the documentation accordingly.